File: TLBB1.5/tabletools/CodeTable/CSharp/Table_ShopNotice.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_ShopNotice : ITableOperate
{ private const string TAB_FILE_DATA = "shop_notice.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_PIC_TYPE,
ID_PIC_SPRITE,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private string m_PicSprite;
 public string PicSprite { get{ return m_PicSprite;}}

private string m_PicType;
 public string PicType { get{ return m_PicType;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_ShopNotice _values = new Tab_ShopNotice();
 _values.m_PicSprite =  valuesList[(int)_ID.ID_PIC_SPRITE] as string;
_values.m_PicType =  valuesList[(int)_ID.ID_PIC_TYPE] as string;

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_ShopNotice.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_ShopNotice implements ITableOperate {
 public final static String TAB_FILE_DATA = "/shop_notice.txt";
 public static final int ID_PIC_TYPE = 0 ;
public static final int ID_PIC_SPRITE = 1 ;
public static final int MAX_RECORD = 2 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private String m_PicSprite;
 public String GetPicSprite() { return m_PicSprite;}

private String m_PicType;
 public String GetPicType() { return m_PicType;}

public boolean LoadTable(HashMap<Integer, Table_ShopNotice> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_ShopNotice> _hash = (HashMap<Integer, Table_ShopNotice> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_ShopNotice _values = new Table_ShopNotice();
 _values.m_PicSprite =  (String)valuesList.get(ID_PIC_SPRITE);
_values.m_PicType =  (String)valuesList.get(ID_PIC_TYPE);

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

